kalite/coachreports/tests/coachreport_tests.py

```diff
@@ -139,3 +141,114 @@ def test_playlist_progress(self):
         self.assertEqual(exercises[0]["title"],"Comparing two-digit numbers")
+
+
+class LearnerLogAPITests(FacilityMixins,
+                       StudentProgressMixin,
+                       CreateZoneMixin,
+                       CreateAdminMixin,
+                       KALiteTestCase):
+    """
+    These tests test the learner log API endpoint in the coachreports app.
+    """
+
+    def setUp(self):
+        super(LearnerLogAPITests, self).setUp()
+
+        self.admin_data = {"username": "admin", "password": "admin"}
+        self.admin = self.create_admin(**self.admin_data)
+
+        self.zone = self.create_zone()
+        self.device_zone = self.create_device_zone(self.zone)
+        self.facility = self.create_facility()
+
+        self.group = self.create_group(name='group1', facility=self.facility)
+        self.empty_group = self.create_group(name='empty_group', facility=self.facility)
+
+        self.facility_user = self.create_student()
+
+        self.topic1 = create({
+            "id": "stuff_here",
+            "slug": "stuff_here",
+            "path": "qwrqweqweqwe",
+            "kind": "Topic",
+            "title": "",
+            "description": "",
+            "available": True,
+        })
+        self.topic2 = create({
+            "id": "stuff_there",
+            "slug": "stuff_there",
+            "path": "qwrqweqweqw",
+            "kind": "Topic",
+            "title": "",
+            "description": "",
+            "available": True,
+        })
+
+        self.exercise1 = create({
+            "id": "stuff_here_ex",
+            "slug": "stuff_here_ex",
+            "path": "qwrqweqweqwe/qwrqweqweq",
+            "kind": "Exercise",
+            "title": "",
+            "description": "",
+            "available": True,
+            "parent": self.topic1,
+        })
+        self.exercise2 = create({
+            "id": "stuff_there_ex",
+            "slug": "stuff_there_ex",
+            "path": "qwrqweqweqw/qwrqweqw",
+            "kind": "Exercise",
+            "title": "",
+            "description": "",
+            "available": True,
+            "parent": self.topic2,
+        })
+
+        self.create_exercise_log(user=self.facility_user, exercise_id=self.exercise1.id)
+        self.create_exercise_log(user=self.facility_user, exercise_id=self.exercise2.id)
+
+    def tearDown(self):
+        self.exercise1.delete()
+        self.exercise2.delete()
+        self.topic1.delete()
+        self.topic2.delete()
+
+    def test_learner_log_topic_filters(self):
+
+        self.client.login(username='admin', password='admin')
+        api_resp_1 = json.loads(self.client.get("%s?facility_id=%s" % (self.reverse("learner_logs"), self.facility.id)).content)
+        api_resp_2 = json.loads(self.client.get("%s?facility_id=%s&topic_ids=%s" % (self.reverse("learner_logs"), self.facility.id, json.dumps([self.topic1.id]))).content)
+        assert len(api_resp_2["contents"]) < len(api_resp_1["contents"])
```

Comment:
Looking good! Still 2 things. Again, happy to merge as-is but I think some small easy improvements could be made.

1. Use the TestCase [assertion methods](https://docs.djangoproject.com/en/1.9/topics/testing/tools/#assertions) for pretty output.
2. Instead of asserting about the *length*, can you make assertions about the *content*? E.g.
```python
self.assertEqual(api_resp_2["contents"], ["foo", "bar"])
```
Or do you think that is likely to change frequently from version to version, and hence maybe *too* specific?